src/XMakeBuildEngine/Construction/ProjectRootElement.cs

```diff
@@ -1714,8 +1714,23 @@ public void Save(Encoding saveEncoding)
                 {
                     using (ProjectWriter projectWriter = new ProjectWriter(_projectFileLocation.File, saveEncoding))
                     {
-                        projectWriter.Initialize(XmlDocument);
-                        XmlDocument.Save(projectWriter);
+                        var xmlWithNoImplicits = (XmlDocument)XmlDocument.Clone();
```

Comment:
At first glance the variable name led me to believe that it wouldn't have any implicit imports when it was cloned, not that you were going to remove them afterwards.  It might be possible to make this clearer with a better name.